### ISSUE TITLE
Corrupt update files when trying to install any game. Ubuntu 20.04

### ISSUE THREAD

**Issue by ptriantafy** [ValveSoftware/steam-for-linux#8854]
Hello,

I've tried installing multiple games on my PC, but when they finish downloading steam states that their update files are corrupt. I deleted steam download cache,  cleared ~/.cache folder, verified integrity of game files, created a new steam library on another drive, reinstalled steam (using .deb file), reinstalled Ubuntu and I even tried different distros. None of them worked.

Any suggestion would be much appreciated, thank you in advance.

**Comment by kisak-valve** [comment]
Hello @PanagiotisTriantafyllopoulos, can you share `~/.steam/root/logs/content_log.txt` as a file or in a [gist](https://gist.github.com)? Also, what filesystem(s) are you using with your Steam library folders?

**Comment by PanagiotisTriantafyllopoulos** [comment]
[content_log.txt](https://github.com/ValveSoftware/steam-for-linux/files/9804073/content_log.txt)

I was using ext4 for all my libraries

**Comment by bellipier** [comment]
@kisak-valve Same issue is happening to me when trying to install or update Apex Legends on Kubuntu 22.04. Using all ext4 disks. 

Thing I have tried: 
- I tried participating in beta and not using the beta. 
- I have tried to reinstall the game multiple times. 
- Tried reinstalling steam. 
- Tried changing download region. 

Btw, the game was working and updating fine for months. Around 1 week ago this issue started to happen.

Sometimes the update deletes all the game files and it starts to download the game from scratch but then fails with "Corrupted update files" after verifying and gets stuck in a loop of verifying and "Corrupted update files" error. I managed to get the game working after a full re-download, copying game files from "downloading" folder into the game folder by force, then using a "appmanifest_xxxx.acf" file which I borrowed from a friend who had the game installed. This, I assume, tricked Steam client into believing the game is installed/downloaded correctly and game worked fine for several days (which can prove the downloaded files were fine/not corrupted and it's a Steam client bug) but a new update for the game got released a couple of days ago and it broke the game files again and I got the "Corrupted update files" error again and now I guess I have to redownload the entire 60GB again and try my ghetto workaround... What could be the culprit here?

**Comment by GrueMaster** [comment]
I'm seeing the same issue on Ubuntu 20.04.  Only thing I can think of is my /home is on an LVM partition (formatted ext4).  I'm going to try another drive in my system w/o LVM and map it to ~/.steam, see if that resolves the issue.

**Comment by angrybacon** [comment]
Similar if not the same issue here

I've had this issue in the past but didn't bother to check the logs at the time. Solved by reinstalling Steam and/or the specific games where it happened.

This time it happens to the 3 games I've tried installing: Dota 2, Don't Starve and Doom Eternal.

Below are the logs corresponding to me clicking the *Resume* download button as the "Corrupt update files" shows up from the download screen for Don't Starve.

<details>
<summary><code>$ tail -f .local/share/Steam/logs/content_log.txt</code></summary>

```
[2022-12-05 15:46:10] AppID 219740 scheduler update : Priority First, not played for 7447390 seconds, update disabled for 0 seconds
[2022-12-05 15:46:10] AppID 219740 state changed : Update Required,Update Queued,Update Optional,Update Started,
[2022-12-05 15:46:10] AppID 219740 state changed : Update Required,Update Queued,Update Optional,Update Running,Update Started,
[2022-12-05 15:46:10] AppID 219740 update changed : Running Update,
[2022-12-05 15:46:10] AppID 219740 update changed : Running Update,Reconfiguring,
[2022-12-05 15:46:10] AppID 219740 update changed : Running Update,
[2022-12-05 15:46:11] AppID 219740 update started : download 1688274464/1689031696, store 0/0, reuse 0/0, delta 0/0, stage 2486144536/2487193112 
[2022-12-05 15:46:11] AppID 219740 update changed : Running Update,Downloading,Staging,
[2022-12-05 15:46:11] Downloading 1 chunks from depot 219742
[2022-12-05 15:46:11] AppID 219740 update changed : Running Update,Staging,
[2022-12-05 15:46:11] AppID 219740 update changed : Running Update,
[2022-12-05 15:46:11] AppID 219740 update changed : Running Update,Verifying,
[2022-12-05 15:46:14] Validation: corrupt chunk 89a51604da607291c97078869bd07c314bd10997 has f27dafc5d94ab1d4c939716d716b7231c0cffa8b at offset 10485760 (1048576 bytes)
[2022-12-05 15:46:14] Validation: 1 chunks corrupt of 14 total in file "data\sound\dontstarve_common.bank"
[2022-12-05 15:46:14] Validation: full scan in "/home/angrybacon/.local/share/Steam/steamapps/downloading/219740" found 1/2977 mismatching files (1.05 MB/1.02 GB) in 2 seconds).
[2022-12-05 15:46:18] AppID 219740 update canceled : Staged file validation failed (0 missing, 1 corrupt, 0 unreadable) (Corrupt update files)
[2022-12-05 15:46:18] AppID 219740 update changed : Running Update,Verifying,Stopping,
[2022-12-05 15:46:18] Corruption detected in '/home/angrybacon/.local/share/Steam', clean bytes tally reset to 0
[2022-12-05 15:46:18] AppID 219740 update changed : Running Update,Stopping,
[2022-12-05 15:46:18] AppID 219740 update changed : None
[2022-12-05 15:46:18] AppID 219740 state changed : Update Required,Update Queued,Update Optional,Update Started, (Corrupt update files)
[2022-12-05 15:46:18] AppID 219740 state changed : Update Required,Update Optional,Update Paused,Update Started,
[2022-12-05 15:46:18] AppID 219740 scheduler finished : removed from schedule (result Corrupt update files, state 0x612)
```
</details>

<details>
<summary><code>$ duf</code></summary>

```
╭──────────────────────────────────────────────────────────────────────────────────────────╮
│ 3 local devices                                                                          │
├────────────┬────────┬───────┬────────┬───────────────────────────────┬──────┬────────────┤
│ MOUNTED ON │   SIZE │  USED │  AVAIL │              USE%             │ TYPE │ FILESYSTEM │
├────────────┼────────┼───────┼────────┼───────────────────────────────┼──────┼────────────┤
│ /          │  19.0G │  8.6G │   9.4G │ [#########...........]  45.3% │ ext4 │ /dev/sda2  │
│ /boot      │ 510.0M │ 59.1M │ 450.8M │ [##..................]  11.6% │ vfat │ /dev/sda1  │
│ /home      │ 896.1G │  7.9G │ 842.6G │ [....................]   0.9% │ ext4 │ /dev/sda3  │
╰────────────┴────────┴───────┴────────┴───────────────────────────────┴──────┴────────────╯
```
</details>

<details>
<summary><code>$ cat /etc/fstab</code></summary>

``` 
# Static information about the filesystems.
# See fstab(5) for details.

# <file system> <dir> <type> <options> <dump> <pass>
# /dev/sda2
UUID=95dc3231-7f7d-4de8-92f0-f6cd0e864aa1	/         	ext4      	rw,relatime	0 1

# /dev/sda1
UUID=4724-2D99      	/boot     	vfat      	rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro	0 2

# /dev/sda3
UUID=194f4f2b-4b0c-4ba6-b896-2b91f152dc82	/home     	ext4      	rw,relatime	0 2
```
</details>

System details as per the in-Steam menu: https://gist.github.com/angrybacon/94e5993a36fc71ec885c51b6323fd011

**Comment by ptriantafy** [comment]
UPDATE: Turns out one of my ram modules was faulty, after I replaced it everything worked fine, I didn't have to reinstall steam or do anything else.

**Comment by angrybacon** [comment]
@ptriantafy Thanks for the heads-up, just to be sure I just finished testing my own and 1 module is completely dead apparently (never had memtest86 scroll that fast). I have no idea whether the fact that it was in the second slot made my machine useable still or if Arch/whatever is in charge of addressing memory is smart enough not rely completely on faulty memory...

I still have the corrupt issue with Dota 2 but it's gone for Don't Starve. I'll reinstall my system completely just in case faulty ram may have compromised other parts of the system (driver installation perhaps?)

**Comment by angrybacon** [comment]
The corrupt issue eventually went away on a fresh install. The game plays nice too, which is something I'll report in another issue I commented on too: https://github.com/ValveSoftware/Dota-2/issues/2184#issuecomment-1333752344

TL;DR if you have corrupt issues while installing games and you can't verify integrity for a given game (nothing is installed yet), and you are sick of reading "just use ext4" over and over, double check your memory

**Comment by samuk10** [comment]
> UPDATE: Turns out one of my ram modules was faulty, after I replaced it everything worked fine, I didn't have to reinstall steam or do anything else.

ty! disabling XMP fixed for me!

I was thinking the problem was a new nvme bought from china! have tried everything about hd fault!

**Comment by Corben78** [comment]
Had an issue with downloads as well, steam ui said corrupted download, the content_log.txt file said upacking chunk failed.
I was able to update some of the games and tools via hosted network transfer though, but not when downloading from the internet.
I changed RAM settings, disabled XMP, did extensive RAM tests without errors, still got corrupted downloads.

Then I found this issue: https://github.com/ValveSoftware/steam-for-linux/issues/12015
I removed all files and folders from my .steam/Steam folder except logs, steamapps, config and userdata.
Steam re-installed itself, now downloads work fine again. Seems like there was a leftover or misconfiguration that interferred. I have this Ubuntu install since 2016 and did always inplace upgrades from LTS to LTS. Just recently these corrupted downloads occured.

**Comment by Muhammad-Fiqri** [comment]
> Had an issue with downloads as well, steam ui said corrupted download, the content_log.txt file said upacking chunk failed. I was able to update some of the games and tools via hosted network transfer though, but not when downloading from the internet. I changed RAM settings, disabled XMP, did extensive RAM tests without errors, still got corrupted downloads.
> 
> Then I found this issue: [#12015](https://github.com/ValveSoftware/steam-for-linux/issues/12015) I removed all files and folders from my .steam/Steam folder except logs, steamapps, config and userdata. Steam re-installed itself, now downloads work fine again. Seems like there was a leftover or misconfiguration that interferred. I have this Ubuntu install since 2016 and did always inplace upgrades from LTS to LTS. Just recently these corrupted downloads occured.

Thanks this solved my problem
